ops/_config.py

```diff
@@ -0,0 +1,268 @@
+                hint = cls._JUJU_TYPES.get(str(hint), 'string')  # type: ignore
+            option['type'] = hint
+            if field.description:  # type: ignore
+                option['description'] = field.description  # type: ignore
+            options[cls._attr_to_juju_name(name)] = option  # type: ignore
+        return {'options': options}
+
+    @classmethod
+    def to_juju_schema(cls) -> dict[str, Any]:
+        """Translate the class to YAML suitable for config.yaml.
+
+        Using :attr:`ConfigBase.to_juju_schema` will generate a YAML schema
+        suitable for use in ``config.yaml``. For example, with the class from
+        the example above::
+
+            print(yaml.safe_dump(MyConfig.to_juju_schema()))
+
+        Will output::
+
+            options:
+                my-bool:
+                    type: boolean
+                    description: A boolean value.
+                my-float:
+                    type: float
+                    default: 3.14
+                    description: A floating point value.
+                my-int:
+                    type: int
+                    default: 42
+                    description: An integer value.
+                my-str:
+                    type: string
+                    default: foo
+                    description: A string value.
+                my-secret:
+                    type: secret
+                    description: A user secret.
+
+        Options with a default value of ``None`` will not have a ``default`` key
+        in the output. If the type of the option cannot be determined, it will
+        be set to ``string``.
+
+        To customise, override this method in your subclass. For example::
+
+            @classmethod
+            def to_juju_schema(cls) -> dict[str, Any]:
+                schema = super().to_juju_schema()
+                # Change the key names to upper-case.
+                schema = {key.upper(): value for key, value in schema.items()}
+                return schema
+        """
+        # Special-case pydantic BaseModel or anything else with a compatible
+        # `model_fields`` attribute.
+        if hasattr(cls, 'model_fields'):
+            return cls.__juju_schema_from_model_fields()
+
+        # Dataclasses, regular classes, etc.
+        attr_docstrings = attrdocs.get_attr_docstrings(cls)
+        options: dict[str, dict[str, bool | int | float | str]] = {}
+        for attr in cls._juju_names():
+            option = {}
+            default = getattr(cls, attr, None)
+            if default is not None and type(default).__name__ in cls._JUJU_TYPES:
+                option['default'] = default
+            option['type'] = cls._attr_to_juju_type(attr, default)
+            doc = attr_docstrings.get(attr)
+            if doc:
+                option['description'] = doc
+            options[cls._attr_to_juju_name(attr)] = option
+        return {'options': options}
+
+
+def generate_juju_schema():
```

Comment:
Note that this is really a placeholder at the moment, until we determine if and how the config.yaml will be generated.